experiment_code/hackrl/experiment.py

```diff
@@ -1105,6 +1208,19 @@ def signal_handler(signum, frame):
     checkpoint_steps = -1
     eval_steps = -1
     eval_step_results = [None, None]
+
+    if TTYREC_ENVPOOL is not None:
+        global EWC_INSTANCE
+        ttyrec_data = TTYREC_ENVPOOL.result()
+        # idx = TTYREC_ENVPOOL.idx
+        ttyrec_data = TTYREC_ENVPOOL.result()
+        idx = TTYREC_ENVPOOL.idx
+        ttyrec_predictions, TTYREC_HIDDEN_STATE[idx] = model(
+            ttyrec_data, TTYREC_HIDDEN_STATE[idx]
+        )
+
+        EWC_INSTANCE = EWC(learner_state.model, TTYREC_ENVPOOL.result())
```

Comment:
I think we want to use more data than one batch. It would be best if you would just pass `TTYREC_ENVPOOL` to `Ewc` and inside called `TTYREC_ENVPOOL.result` few times (or more) with for loop.

Reply:
On average each trajectory in dataset has 10000 steps. Assuming that we need around 300 batches just to ensure we represent whole trajectories with this loss. I'd say we need around 1000 batches for representative sample size.

Reply:
Maybe it's worth to precompute this?